### ISSUE TITLE
no-var-keyword inside a loop

### ISSUE THREAD

**Issue by michaelaird** [palantir/tslint#873]
I'm running into an issue with the no-var-keyword rule.

If you declare a variable inside a loop with let and then try to reference it in an arrow function, TypeScript complains with the following error:

Loop contains block-scoped variable 'existingModal' referenced by a function in the loop. This is only supported in ECMAScript 6 or higher.

here's a code snippet that produces this:

```ts
for (let i: number = 0; i < modals.length; i++)
{
    let existingModal: JQuery = $(modals[i]);

    let preventClose: (eventObject: JQueryEventObject) => any = (hiddenEvent: JQueryEventObject) =>
    {
        hiddenEvent.preventDefault();
    };

    existingModal.bind('hide', preventClose);

    restoreModals[i] = (hideEvent: JQueryEventObject) =>
    {
        existingModal.off('hide', preventClose);
    };
}
```

**Comment by adidahiya** [comment]
1. Heads up, this will be supported in TS 1.8 https://github.com/Microsoft/TypeScript/issues/3915
2. I'd suggest converting the for loop to a `forEach` to force a new scope to be created:

  ```ts
  modals.forEach((modal, i) =>
  {
      ...
  }
  ```
3. I don't think there's any change to the `no-var-keyword` rule required here. This doesn't seem like a common / encouraged pattern, so if you need to use `var`, you should use the inline enable/disable comment flags to disable the rule.

**Comment by michaelaird** [comment]
Awesome! Thanks for the quick reply. I refactored the code to match your suggestion #2 and that did the trick.
